lesson3/hometask03_ankarut.rb

```diff
@@ -0,0 +1,54 @@
+require 'rspec'
+require 'rspec/expectations'
+include RSpec::Matchers
+
+class Array
+
+  def custom_each(obj_class=nil, &block)
+    i = 0
+    while i < self.length
+      el = self[i]
+      i+=1
+      yield(el) if block_given?
+    end
+    if obj_class
+      obj_class[self]
+    else
+      self
+    end
+  end
+end
+
+class Hash
+  def custom_each(&block)
+    if block_given?
+      Array.new(self.to_a).custom_each(self.class, &block)
+    else
+      self
+    end
+  end
+end
+
+
+expect([4,6,8].custom_each { |x| puts x**x }).to eq([4,6,8].each { |x| puts x**x })
```

Comment:
take look on [these samples](https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/yield-matchers) if you want to do it properly